src/robotFactory.js

```diff
@@ -1,17 +1,67 @@
 'use strict';
 
-function BaseRobot() {
-  // implement
+function BaseRobot(name, weight, coords, chipVersion) {
+  this.name = name;
+  this.weight = weight;
+  this.coords = { ...coords };
```

Comment:
```suggestion
  this.coords = coords;
```
не совсем понимаю, зачем тут спред-оператор?

Reply:
Изначально использовал для того чтою просто получить копии ключей